### PR TITLE
[test] [sagemaker] Change traceback output from stdout to stderr

### PR DESCRIPTION
**Pull request by saimidu**
*Issue #, if available:*

## Checklist
- [x] I've prepended PR tag with frameworks/job this applies to : [mxnet, tensorflow, pytorch] | [ei/neuron] | [build] | [test] | [benchmark] | [ec2, ecs, eks, sagemaker]
- [x] (If applicable) I've documented below the DLC image/dockerfile this relates to
- [x] (If applicable) I've documented below the tests I've run on the DLC image
- [x] (If applicable) I've reviewed the licenses of updated and new binaries and their dependencies to make sure all licenses are on the Apache Software Foundation Third Party License Policy Category A or Category B license list.  See [https://www.apache.org/legal/resolved.html](https://www.apache.org/legal/resolved.html).
- [x] (If applicable) I've scanned the updated and new binaries to make sure they do not have vulnerabilities associated with them.

## Reviewer Checklist
* For reviewer, before merging, please cross-check:
- [ ] I've verified the code change on the config file mentioned above has already been reverted

*Description:*
Change the traceback printed on SageMaker test failure to be obtained from stderr instead of stdout, because stdout is empty.

*Tests run:*
N/A

*DLC image/dockerfile:*
N/A

*Additional context:*
N/A

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license. I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: test/test_utils/sagemaker.py [hunk 1]
```diff
@@ -274,7 +274,7 @@ def execute_sagemaker_remote_tests(image):
             if res.failed:
                 raise DLCSageMakerRemoteTestFailure(
                     f"{pytest_command} failed with error code: {res.return_code}\n"
-                    f"Traceback:\n{res.stdout}"
+                    f"Traceback:\n{res.stderr}"
```
Comment (arjkesh) [inline test/test_utils/sagemaker.py:277]: nit: instead of 'traceback', how about just dump stdout and stderr directly and label them as such?

```
                raise DLCSageMakerRemoteTestFailure(
                    f"{pytest_command} failed with error code: {res.return_code}\n"
                    f"stdout:\n{res.stdout}\n"
                    f"stderr:\n{res.stderr}")
```

Comment (saimidu) [reply test/test_utils/sagemaker.py:277]: Cool.

**Review by arjkesh — APPROVED** [review]

File: test/test_utils/sagemaker.py [hunk 2]
```diff
@@ -308,7 +308,8 @@ def execute_sagemaker_remote_tests(image):
             if res.failed:
                 raise DLCSageMakerRemoteTestFailure(
                     f"{pytest_command} failed with error code: {res.return_code}\n"
-                    f"Traceback:\n{res.stdout}"
+                    f"stdout:\n{res.stdout}\n"
```
Comment (anankira) [inline test/test_utils/sagemaker.py:311]: @saimidu did you mean to dump to stdout and stderr?
